From 6c6ed5ecb8a8dec5cfe1b135f7ba45928e389aed Mon Sep 17 00:00:00 2001
From: Scott Kovach <dskovach@gmail.com>
Date: Wed, 12 Nov 2014 10:40:56 -0800
Subject: [PATCH] added derived Eq,Ord instances to NodeInfo

---
 src/Language/C/Data/Node.hs | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/src/Language/C/Data/Node.hs b/src/Language/C/Data/Node.hs
index f6b28fd..c393263 100644
--- a/src/Language/C/Data/Node.hs
+++ b/src/Language/C/Data/Node.hs
@@ -26,7 +26,7 @@ import Data.Generics
 -- | Parsed entity attribute
 data NodeInfo = OnlyPos  Position {-# UNPACK #-} !PosLength        -- only pos and last token (for internal stuff only)
               | NodeInfo Position {-# UNPACK #-} !PosLength !Name  -- pos, last token and unique name
-           deriving (Data,Typeable)
+           deriving (Data,Typeable,Eq,Ord)
 
 instance Show NodeInfo where
     showsPrec d (OnlyPos p l) =
@@ -35,16 +35,16 @@ instance Show NodeInfo where
       (showString "(NodeInfo ") . (showsPrec d p) . (showString " ") . (showsPrec d l) . (showString " ") . (showsPrec d n) . (showString ")")
 
 -- name equality of attributes, used to define (name) equality of objects
-instance Eq NodeInfo where
-  (NodeInfo   _ _ id1) == (NodeInfo   _ _ id2) = id1 == id2
-  _               == _               =
-    error "Attributes: Attempt to compare `OnlyPos' attributes!"
+--instance Eq NodeInfo where
+--  (NodeInfo   _ _ id1) == (NodeInfo   _ _ id2) = id1 == id2
+--  _               == _               =
+--    error "Attributes: Attempt to compare `OnlyPos' attributes!"
 
 -- attribute ordering
-instance Ord NodeInfo where
-  (NodeInfo   _ _ id1) <= (NodeInfo   _ _ id2) = id1 <= id2
-  _               <= _               =
-    error "Attributes: Attempt to compare `OnlyPos' attributes!"
+--instance Ord NodeInfo where
+--  (NodeInfo   _ _ id1) <= (NodeInfo   _ _ id2) = id1 <= id2
+--  _               <= _               =
+--    error "Attributes: Attempt to compare `OnlyPos' attributes!"
 
 instance Pos NodeInfo where
   posOf (OnlyPos pos _) = pos
-- 
1.9.3 (Apple Git-50)

